app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -54,26 +71,65 @@ def latest_reports?
       latest_reports.limit(1).present?
     end
 
-    def host_reports_translated_header(shortname, longname)
-      "<th style='width:85px; class='ca'><span class='small' title='' data-original-title='#{longname}'>#{shortname}</span></th>"
+    def failure_hosts_query(format)
+      "report_failures > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
     end
 
-    private
+    def failure_hosts(format)
+      Host.search_for(failure_hosts_query(format)).reorder("").count
+    end
 
-    def disabled_hosts(_format = nil)
-      Host.authorized(:view_hosts, Host).where(:enabled => false).count
+    def change_hosts_query(format)
+      "report_failures = 0 and report_changes > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
```

Comment:
I see that in `failure_hosts_query` you are just referring to report_failures
in ```change_hosts_query``` you have decided to include report_failures and report_changes
and in ```nochange_hosts_query``` you referred to all three of them 
What is the idea to do so? why not include all three of them in each query with the right values?

Reply:
What are the "right values"? Can you provide examples?

Reply:
What I meant is, Why did you omit some of the conditions in different functions?  
```ruby 
def change_hosts_query(format)
      "report_failures = 0 and report_changes > 0 and  report_nochanges = 0  and  report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"

 def failure_hosts_query(format)
      "report_failures > 0  and report_changes = 0 and report_nochanges = 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
    end

def nochange_hosts_query(format)
      "report_failures = 0 and report_changes = 0 and report_nochanges > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
    end
```